include/ggml/ggml.h

```diff
@@ -2342,11 +2343,13 @@ extern "C" {
     GGML_API const void * gguf_get_arr_data(const struct gguf_context * ctx, int key_id);
     GGML_API const char * gguf_get_arr_str (const struct gguf_context * ctx, int key_id, int i);
 
-    GGML_API int            gguf_get_n_tensors    (const struct gguf_context * ctx);
-    GGML_API int            gguf_find_tensor      (const struct gguf_context * ctx, const char * name);
-    GGML_API size_t         gguf_get_tensor_offset(const struct gguf_context * ctx, int i);
-    GGML_API char *         gguf_get_tensor_name  (const struct gguf_context * ctx, int i);
-    GGML_API enum ggml_type gguf_get_tensor_type  (const struct gguf_context * ctx, int i);
+    GGML_API int            gguf_get_n_tensors      (const struct gguf_context * ctx);
+    GGML_API int            gguf_find_tensor        (const struct gguf_context * ctx, const char * name);
+    GGML_API size_t         gguf_get_tensor_offset  (const struct gguf_context * ctx, int i);
+    GGML_API char *         gguf_get_tensor_name    (const struct gguf_context * ctx, int i);
+    GGML_API enum ggml_type gguf_get_tensor_type    (const struct gguf_context * ctx, int i);
+    GGML_API size_t         gguf_get_tensor_size    (const struct gguf_context * ctx, int i);
+    GGML_API int            gguf_find_and_get_tensor(const struct gguf_context * ctx, const char * name, char ** data, size_t * size);
```

Comment:
This is looking better, but still needs some work. Neither of the changes to `gguf` are needed, so try to avoid them and do the same using the existing API. The final PR should not contain any modifications to `ggml.h`. The only one that can remain is the `gguf_get_tensor_size()` helper function